src/core.jl

```diff
@@ -109,6 +118,66 @@ struct Polygon <: Drawable
     vertices::Vector{Point}
 end
 
+"""
+    BoundaryFill <: AbstractPolyFillAlgorithm
+    BoundaryFill(x::Int, y::Int, fill_value::T, boundary_value::T)
+
+    draw(img, verts, alg::BoundaryFill; closed)
+    draw!(img, verts, alg::BoundaryFill; closed)
+
+Applies boundary filling algortithm on image inside the vertices provided in order
+
+# Output
+
+Return the boundary filled image inside the vertices(provided in order). 
+
+# Arguments
+
+## `x` && `y`
+
+Boundary fill is a seeded polygon filling algorithm. 
+So we need to provide the seed point (x,y) inside the image from where the algorithm can start its function.
+// To-do checking if verts and seed are inside the image or note
+
+## `fill_value`
+
+The pixels inside the boundary of vertices of polygon are filled with this colorant parameter.
+
+## `boundary_value`
+
+The pixels to define the boundary using verts is filled with this colorant parameter. 
+Without this parameter, the whole picture will be filled with the `fill_value`
```

Comment:
Is it? Add a test to make sure this is correct.

```diff
- BoundaryFill(;x::Int = 0, y::Int = 0, fill_value::Colorant = RGB(1), boundary_value::Colorant = RGB(1)) = 
+ BoundaryFill(;x::Int = 0, y::Int = 0, fill_value::Colorant = RGB(1), boundary_value::Colorant = fill_value) =
  BoundaryFill(x, y, fill_value, boundary_value)
```

Reply:
Sure